hadrons/parallel/continuous_beam_numba_parallel.py

```diff
@@ -0,0 +1,323 @@
+
+    '''
+    find a time step dt which fulfils the von Neumann criterion, i.e. ensures the numericl error does not increase but
+    decreases and eventually damps out
+    '''
+    dt = 1.
+    von_neumann_expression = False
+    Efield_V_cm = voltage_V/d_cm # to ensure the same dt in all simulations
+    start_von_neumann = time()
+    while not von_neumann_expression:
+        dt /= 1.01
+        # as defined in the Deghan (2004) paper
+        sx = ion_diff*dt/(unit_length_cm**2)
+        sy = ion_diff*dt/(unit_length_cm**2)
+        sz = ion_diff*dt/(unit_length_cm**2)
+        cx = cy = 0.0
+        cz = ion_mobility*Efield_V_cm*dt/unit_length_cm
+        # check von Neumann's criterion
+        von_neumann_expression = (2*(sx + sy + sz) + cx**2 + cy**2 + cz**2 <= 1 and cx**2*cy**2*cz**2 <= 8*sx*sy*sz)
+    stop_von_neumann = time()
+    print(f"Execution for Von Neuman expression: {stop_von_neumann-start_von_neumann}s")
+    '''
+    Calculate the number of step required to drag the two charge carrier distributions apart
+    along with the number of tracks to be uniformly distributed over the domain
+    '''
+    separation_time_steps = int(d_cm/(2.*ion_mobility*Efield_V_cm*dt))
+    n_separation_times = 3
+    computation_time_steps = separation_time_steps * n_separation_times
+    simulation_time_s = computation_time_steps*dt
+    number_of_tracks = int(fluence_cm2_s* simulation_time_s*area_cm2)
+    if number_of_tracks < 1:
+        number_of_tracks = 1 # initial recombination
+
+    # Gaussian track structure
+    N0 = LET_eV_cm/W                 # Linear charge carrier density
+    b_cm = track_radius              # Gaussian track radius
+    Gaussian_factor = N0/(pi*b_cm**2)
+
+    # preallocate arrays
+    x_coordinates_ALL = np.random.uniform(0, 1, int(number_of_iterations))*no_xy
+    y_coordinates_ALL = np.random.uniform(0, 1, int(number_of_iterations))*no_xy
+    positive_array = np.zeros((no_xy, no_xy, no_z_with_buffer))
+    negative_array = np.zeros((no_xy, no_xy, no_z_with_buffer))
+    positive_array_temp = np.zeros((no_xy, no_xy, no_z_with_buffer))
+    negative_array_temp = np.zeros((no_xy,  no_xy, no_z_with_buffer))
+    no_recombined_charge_carriers = 0.0
+    no_initialised_charge_carriers = 0.0
+    positive_array_temp = np.zeros((no_xy, no_xy, no_z_with_buffer))
+    negative_array_temp = np.zeros((no_xy, no_xy, no_z_with_buffer))
+
+    '''
+    The following part is only used to plot the figure; consider to remove
+    '''
+    MINVAL = 0.
+    MAXVAL = 0.
+    if SHOW_PLOT:
+        import matplotlib.gridspec as gridspec
+        import matplotlib.pyplot as plt
+
+        plt.close('all')
+        fig = plt.figure()
+        plt.ion()
+        gridspec.GridSpec(1, 3, width_ratios=[1, 1, 1])
+        # define locations of subfigures in the grid
+        ax1 = plt.subplot2grid((5,3),(0,0),rowspan=4)
+        ax2 = plt.subplot2grid((5,3),(0,1),rowspan=4)
+        ax3 = plt.subplot2grid((5,3),(0,2),rowspan=4)
```

Comment:
branching may affect performance of parallel exeuction, please simply remove that code